_targets/riscv64/noelv/build.project

```diff
@@ -57,20 +57,57 @@ b_build_target() {
 	image_builder.py script --nvm "$NVM_CONFIG" --script "$PLO_SCRIPT_PREINIT" --out script.plo
 	RAM_SCRIPT=1 image_builder.py script --nvm "$NVM_CONFIG" --script "$PLO_SCRIPT_PREINIT" --out script-ram.plo
 	make -C plo all
-	cp -a "${PREFIX_PROG_STRIPPED}plo-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.img" "$PREFIX_BOOT/plo.img"
 
 	b_log "Adding PLO to Phoenix SBI"
-	make -C plo/riscv-sbi all FDT_PATH="$PREFIX_PROJECT/_projects/$TARGET/$TARGET.dtb" \
-		PAYLOAD_PATH="${PREFIX_PROG_STRIPPED}plo-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.img" PAYLOAD_ADDR=0x200000
+
+	make -C plo/riscv-sbi ram FDT_PATH="$PREFIX_PROJECT/_projects/$TARGET/$TARGET.dtb" \
+		PAYLOAD_PATH="${PREFIX_PROG_STRIPPED}plo-ram-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.img" PAYLOAD_ADDR=0x10000
+
+	make -C plo/riscv-sbi base FDT_PATH="$PREFIX_PROJECT/_projects/$TARGET/$TARGET.dtb" \
+		PAYLOAD_PATH="${PREFIX_PROG_STRIPPED}plo-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.img" PAYLOAD_ADDR=0x10000
+
+	cp -a "${PREFIX_PROG_STRIPPED}sbi-${TARGET_SUBFAMILY}.img" "$PREFIX_BOOT/sbi-${TARGET_SUBFAMILY}.img"
+	cp -a "${PREFIX_PROG_STRIPPED}sbi-${TARGET_SUBFAMILY}.img" "$PREFIX_BOOT/sbi-ram-${TARGET_SUBFAMILY}.img"
 }
 
 
 b_image_target() {
-
 	image_builder.py -v partition --nvm "$NVM_CONFIG" --name kernel --script "$PLO_SCRIPT_USER"
 
+	b_log "Creating rootfs"
+
+	ROOTFS="$PREFIX_BOOT/part_rootfs.img"
+
+	local erase_sz
+	erase_sz=$(image_builder.py query --nvm "$NVM_CONFIG" '{{ nvm.flash0._meta.block_size }}')
+	# Create jffs2 image with padding at the end of the final erase block, no cleanmarkers (are written by PLO)
+	mkfs.jffs2 -U -l -m none -e "$erase_sz" -s "$SIZE_PAGE" -r "$PREFIX_ROOTFS/" -o "$ROOTFS" -p -n
+
+	if sumtool -n -e "$erase_sz" -l -i "$ROOTFS" -o "$ROOTFS.tmp" 2> /dev/null; then
+		echo "JFFS2 Summary nodes created"
+		mv "$ROOTFS.tmp" "$ROOTFS"
+	fi
+
+	local FS_OFFS
+	FS_OFFS=$(image_builder.py query --nvm "$NVM_CONFIG" '{{ nvm.flash0.rootfs.offs }}')
+	local FS_SZ
+	FS_SZ=$(image_builder.py query --nvm "$NVM_CONFIG" '{{ nvm.flash0.rootfs.size }}')
+	sz=$(du -b "$ROOTFS" | awk '{ print $1 }')
+	echo "rootfs size: $((sz / 1024)) kB / $((FS_SZ / 1024)) kB"
+	[ "$sz" -gt "$FS_SZ" ] && b_die "rootfs size exceeds the target partition!"
+
+	local start=$(((FS_OFFS+sz+erase_sz-1)/erase_sz))  # Start clean block after rootfs
+	local end=$(((FS_OFFS+FS_SZ+erase_sz-1)/erase_sz)) # Number of blocks up to rootfs end
+	local nblocks=$((end-start-1))                     # Clean blocks
+
+	# cmd for creating cleanmarkers in plo
+	echo "Format jffs2 partition in plo:"
+	echo "jffs2 -d 2.0 -c ${start}:${nblocks}:${erase_sz}:16"
```

Comment:
note: should be harmless to format full partition (starting at `FS_OFFS`), cleanmarkers should be ignored by jffs if first page of erase block is non-empty - I assume this is done for optimization?

Edit: IMHO you should always erase the full partition - some NOR plo drivers assume the underlying device was cleared (if not - they will do partial programming only)